improver_tests/threshold/test_ThresholdPytest.py

```diff
@@ -0,0 +1,599 @@
+            ),  # 2x2 realization 1
+            np.array([[1.0, 0], [0, 0]], dtype=np.float32),
+        ),
+        # Vicinity processing, with one central value exceeding the threshold
+        # resulting in the whole domain returning probabilities of 1
+        (
+            {"threshold_values": 0.5, "vicinity": 3000},
+            1,
+            np.array([[0, 0, 0], [0, 1.0, 0.0], [0, 0, 0]]),
+            np.ones((3, 3), dtype=np.float32),
+        ),
+        # Vicinity processing, with one corner value exceeding the threshold
+        # resulting in neighbourhing cells probabilities of 1 within the limit
+        # of the defined vicinity radius (2x2km grid cells)
+        (
+            {"threshold_values": 0.5, "vicinity": 3000},
+            1,
+            np.array([[1.0, 0, 0], [0, 0, 0.0], [0, 0, 0]]),
+            np.array([[1.0, 1.0, 0], [1.0, 1.0, 0.0], [0, 0, 0]], dtype=np.float32),
+        ),
+        # Vicinity processing, with one masked value exceeding the threshold.
+        # This masked point is not considered, and so zeros are returned
+        # at neighbouring points within the vicinity radius. The masking is
+        # preserved in the resulting thresholded data.
+        (
+            {"threshold_values": 0.5, "vicinity": 3000},
+            1,
+            np.ma.masked_array(
+                [[0, 2.0, 0], [0, 0, 0.0], [0, 0, 0]],
+                mask=[[0, 1, 0], [0, 0, 0], [0, 0, 0]],
+            ),
+            np.ma.masked_array(
+                [[0, 0, 0], [0, 0, 0], [0, 0, 0]],
+                mask=[[0, 1, 0], [0, 0, 0], [0, 0, 0]],
+                dtype=np.float32,
+            ),
+        ),
+        # multiple thresholds and multiple vicinities; input a 5x5 grid
+        # with value 1 at the top-left point. For the first threshold
+        # probabilities of 1 are returned in the top left corner over
+        # 2x2 cells for the first vicinity, and over 3x3 cells for the
+        # second vicinity. The second threshold returns only 0 probabilities.
+        (
+            {"threshold_values": [0.5, 1.5], "vicinity": [3000, 5000]},
+            1,
+            np.r_[[1], [0] * 24].reshape((5, 5)),
+            np.stack(
+                [
+                    np.stack(
+                        [
+                            np.r_[[1] * 2, [0] * 3, [1] * 2, [0] * 18]
+                            .reshape((5, 5))
+                            .astype(np.float32),
+                            np.r_[[1] * 3, [0] * 2, [1] * 3, [0] * 2, [1] * 3, [0] * 12]
+                            .reshape((5, 5))
+                            .astype(np.float32),
+                        ]
+                    ),
+                    np.stack(
+                        [
+                            np.zeros((5, 5), dtype=np.float32),
+                            np.zeros((5, 5), dtype=np.float32),
+                        ]
+                    ),
+                ]
+            ),
```

Comment:
Move all the `astype` bits to the outside, redefine the second inner stack, and this looks much prettier:
```suggestion
            np.stack(
                [
                    np.stack(
                        [
                            np.r_[[1] * 2, [0] * 3, [1] * 2, [0] * 18].reshape((5, 5)),
                            np.r_[
                                [1] * 3, [0] * 2, [1] * 3, [0] * 2, [1] * 3, [0] * 12
                            ].reshape((5, 5)),
                        ]
                    ),
                    np.zeros((2, 5, 5)),
                ]
            ).astype(np.float32),
```